### ISSUE TITLE
[Docker Agent 1.17] After Containers rotates logs datadog agents stop send logs

### ISSUE THREAD

**Issue by cesarcelism** [DataDog/datadog-agent#4886]
**Output of the info page (if this is a bug)**
```
===============
Agent (v7.17.0)
===============

  Status date: 2020-02-12 14:36:14.226983 UTC
  Agent start: 2020-02-11 21:55:50.611689 UTC
  Pid: 345
  Go Version: go1.12.9
  Python Version: 3.7.6
  Build arch: amd64
  Check Runners: 4
  Log Level: debug

  Paths
  =====
    Config File: /etc/datadog-agent/datadog.yaml
    conf.d: /etc/datadog-agent/conf.d
    checks.d: /etc/datadog-agent/checks.d

  Clocks
  ======
    System UTC time: 2020-02-12 14:36:14.226983 UTC

  Host Info
  =========
    bootTime: 2020-02-11 01:50:39.000000 UTC
    kernelVersion: 4.19.95-coreos
    os: linux
    platform: debian
    platformFamily: debian
    platformVersion: bullseye/sid
    procs: 71
    uptime: 20h5m14s
    virtualizationRole: guest
    virtualizationSystem: docker

  Hostnames
  =========
    ec2-hostname: ip-*-*-*-*.us-west-2.compute.internal
    hostname: i-08c****3b
    instance-id: i-08c********3b
    socket-fqdn: 48******b
    socket-hostname: ***b***
    host tags:
      docker_swarm_node_role:worker
    hostname provider: aws
    unused hostname providers:
      configuration/environment: hostname is empty
      gce: unable to retrieve hostname from GCE: status code 404 trying to GET http://169.254.169.254/computeMetadata/v1/instance/hostname

  Metadata
  ========
    cloud_provider: AWS
    hostname_source: aws

=========
Collector
=========

  Running Checks
  ==============

    cpu
    ---
      Instance ID: cpu [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/cpu.d/conf.yaml.default
      Total Runs: 4,002
      Metric Samples: Last Run: 6, Total: 24,006
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s


    disk (2.6.0)
    ------------
      Instance ID: disk:e5dffb8bef24336f [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/disk.d/conf.yaml.default
      Total Runs: 4,001
      Metric Samples: Last Run: 194, Total: 776,194
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 71ms


    docker
    ------
      Instance ID: docker [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/docker.d/conf.yaml.default
      Total Runs: 4,001
      Metric Samples: Last Run: 178, Total: 648,640
      Events: Last Run: 0, Total: 18
      Service Checks: Last Run: 1, Total: 4,001
      Average Execution Time : 60ms


    file_handle
    -----------
      Instance ID: file_handle [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/file_handle.d/conf.yaml.default
      Total Runs: 4,001
      Metric Samples: Last Run: 5, Total: 20,005
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s


    io
    --
      Instance ID: io [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/io.d/conf.yaml.default
      Total Runs: 4,001
      Metric Samples: Last Run: 117, Total: 468,036
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 1ms


    load
    ----
      Instance ID: load [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/load.d/conf.yaml.default
      Total Runs: 4,001
      Metric Samples: Last Run: 6, Total: 24,006
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s


    memory
    ------
      Instance ID: memory [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/memory.d/conf.yaml.default
      Total Runs: 4,001
      Metric Samples: Last Run: 17, Total: 68,017
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s


    network (1.14.0)
    ----------------
      Instance ID: network:e0204ad63d43c949 [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/network.d/conf.yaml.default
      Total Runs: 4,002
      Metric Samples: Last Run: 31, Total: 124,062
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 3ms


    ntp
    ---
      Instance ID: ntp:d884b5186b651429 [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/ntp.d/conf.yaml.default
      Total Runs: 67
      Metric Samples: Last Run: 0, Total: 0
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 1, Total: 67
      Average Execution Time : 20.017s


    uptime
    ------
      Instance ID: uptime [OK]
      Configuration Source: file:/etc/datadog-agent/conf.d/uptime.d/conf.yaml.default
      Total Runs: 4,001
      Metric Samples: Last Run: 1, Total: 4,001
      Events: Last Run: 0, Total: 0
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 0s

========
JMXFetch
========

  Initialized checks
  ==================
    no checks

  Failed checks
  =============
    no checks

=========
Forwarder
=========

  Transactions
  ============
    CheckRunsV1: 4,001
    Dropped: 0
    DroppedOnInput: 0
    Events: 0
    HostMetadata: 0
    IntakeV1: 446
    Metadata: 0
    Requeued: 0
    Retried: 0
    RetryQueueSize: 0
    Series: 0
    ServiceChecks: 0
    SketchSeries: 0
    Success: 8,448
    TimeseriesV1: 4,001

  API Keys status
  ===============
    API key ending with a2379: API Key valid

==========
Endpoints
==========
  https://app.datadoghq.com - API Key ending with:
      - a2379

==========
Logs Agent
==========
    Sending uncompressed logs in SSL encrypted TCP to agent-intake.logs.datadoghq.com on port 10516
    BytesSent: 7.39424441e&#43;08
    EncodedBytesSent: 7.39424441e&#43;08
    LogsProcessed: 1.047821e&#43;06
    LogsSent: 1.047821e&#43;06

  container_collect_all
  ---------------------
    Type: docker
    Status: OK
    Inputs: 457e7b8b6fb1cff5300315ba08dbc2414e85a2533bbe8d753994a86605422858 86ab21e54f8d431dc55adeca261b4821e4b92e1ab6ec76d86e866f889b276262 107b6b0e065424c10ad9d7c362722628c16db711bc38b4870faf780ade522618 a3f92f59e858c8486ff421750f6e3acaa7064a65c70a31c685cb9e49dda0399b 4b044fd1fc5bb660e000d3bbb84df02c48f4aaf6ef9e4833a624b14a22427198

=========
Aggregator
=========
  Checks Metric Sample: 2.2 M
  Dogstatsd Metric Sample: 16,006
  Event: 19
  Events Flushed: 19
  Number Of Flushes: 4,001
  Series Flushed: 1.6 M
  Service Check: 40,146
  Service Checks Flushed: 44,142

=========
DogStatsD
=========
  Event Packets: 0
  Event Parse Errors: 0
  Metric Packets: 16,005
  Metric Parse Errors: 0
  Service Check Packets: 0
  Service Check Parse Errors: 0
  Udp Bytes: 836,257
  Udp Packet Reading Errors: 0
  Udp Packets: 16,006
  Uds Bytes: 0
  Uds Origin Detection Errors: 0
  Uds Packet Reading Errors: 0
  Uds Packets: 0
```

**Describe what happened:**

When the error: `Error streaming logs: unexpected EOF container=3ef3590c8d13bf72aed418445af9f66fb5871ac0921e841b5117dcf325e260f4` (in the app container) occurs in the docker daemon logs, the datadog agent send to datadog the line `grabbing logs: unexpected EOF` and after that stop send more logs, for that container, until the container restarts and/or update the related stack. 

The agent logs don’t show errors (even with debug log level) and the app container logs keep coming through without issues.

**Describe what you expected:**

The datadog agent keep sending logs from the container

**Steps to reproduce the issue:** 

In at docker cluster swarm with json-file with json logs when rotates the log file in the container app and in the docker daemon logs shows 'Error streaming logs: unexpected EOF' 

**Additional environment details (Operating System, Cloud provider, etc):**

* AWS with EC2 instances
* Docker CE 18.06 (docker swarm)
* CoreOS 2303.4.0
* Docker Datadog Agent (1.17)

**Comment by olivielpeau** [comment]
Hi @cesarcelism, thanks for reporting this issue. We've had other reports of this issue as well, and we're currently working on a fix (#4860).

Could you share your full docker log settings as well (in particular `max-file` and `max-size`)? If it contains information you would not want to share in a public issue, feel free to email it to support@datadoghq.com and refer to this issue in your email.

I'll update this issue when we have an image that ships a fix.

**Comment by cesarcelism** [comment]
Thanks for the answer

The compose files from the **app** and the docker **datadog-agent** setup:

```
version: '3.4'

services:
  app:
    image: "${DOCKER_IMAGE}:${DOCKER_TAG}"
    environment:
      DB_USER: ${SERVICE_LIVE_DB_USER}
      DB_PASS: ${SERVICE_LIVE_DB_PASSWORD}
    deploy:
      replicas: ${SERVICE_APP_REPLICAS}
      labels:
        - "traefik.frontend.rule=Host:${APP_DOMAIN}"
        - "traefik.port=${PORT}"
    networks:
      - webgateway
      - bridge
    logging:
      driver: json-file
      options:
        max-size: "100m"

networks:
  webgateway:
    external: true
  bridge:
    external: true
```

datadog agent

```
version: '3.2'

services:

  datadog:
    image: datadog/agent:7.17.0
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /proc/:/host/proc/:ro
      - /sys/fs/cgroup/:/host/sys/fs/cgroup:ro
      - logs:/opt/datadog-agent/run
    deploy:
      mode: global
    environment:
      DD_API_KEY:${DD_API_KEY}
      DD_PROCESS_AGENT_ENABLED: "true"
      DD_DOGSTATSD_NON_LOCAL_TRAFFIC: "false"
      DD_LOGS_ENABLED: "true"
      DD_LOG_LEVEL: "debug"
      DD_LOGS_CONFIG_CONTAINER_COLLECT_ALL: "true"
      DD_AC_EXCLUDE: "image:datadog.*"
    logging:
      driver: json-file
      options:
        max-size: "100m"
```

**Comment by olivielpeau** [comment]
Thanks for sharing this @cesarcelism, based on your setup https://github.com/DataDog/datadog-agent/pull/4860 should indeed fix the bug you're experiencing, I'll keep you posted when an image that includes a fix is available.

**Comment by cesarcelism** [comment]
I'm testing in my pre-production environment with **1.18.0-rc.1** without issues in the last 2 days 😃 . I'll be waiting for the 1.18.0 to upgrade my production environment.

Thanks!

**Comment by cesarcelism** [comment]
problem resolved at 7.20.0 🙇🏽
